qiskit_machine_learning/neural_networks/effective_dimension.py

```diff
@@ -0,0 +1,349 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An implementation of the effective dimension algorithm."""
+
+import time
+from typing import Optional, Union, List, Callable, Tuple
+
+import numpy as np
+from scipy.special import logsumexp
+
+from qiskit.utils import algorithm_globals
+
+from .opflow_qnn import OpflowQNN
+from .neural_network import NeuralNetwork
+
+
+class EffectiveDimension:
+    """
+    This class computes the global effective dimension for Qiskit
+    :class:`~qiskit_machine_learning.neural_networks.NeuralNetwork`s
+    following the definition used in [1].
+
+    **References**
+        [1]: Abbas et al., The power of quantum neural networks. `The power of QNNs
+            <https://arxiv.org/pdf/2011.00027.pdf>`_
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        params: Optional[Union[List[float], np.ndarray, int]] = 1,
+        inputs: Optional[Union[List[float], np.ndarray, int]] = 1,
```

Comment:
These no longer need to be Optional as None is not a valid value any more for these right?